### PR TITLE
[sc-37186] Fix tooltip formatting callback

### PR DESCRIPTION
**Pull request by wmai**
## Summary

The goal for this PR is to fix charts tooltip formatting callback.

(Internal for Opendatasoft only) Associated Shortcut ticket: [sc-37186](https://app.shortcut.com/opendatasoft/story/37186).

### Changes

- **Fix tooltip callback:** previously, we provided the `dataIndex` to the callback, while we should have directly passed [the parsed data to be formatted](https://github.com/opendatasoft/ods-dataviz-sdk/pull/157/files#diff-f6428828babc3979ce4e60126a99390b9ef29f10d095dc0aff101a9b9b050c43R89-R130).
- **Standardize the callback name:** formerly named `tooltip.label`, renamed to `tooltip.numberFormatter`.
- **Fix style:** data labels position and opacity.
- **Pie charts tooltips:** Pie charts are different, we need to display `x` as the label (other charts display the series name). 
- **Test:** a Storybook story has been created to test and visualize the impact of the locale change on the tooltips. 

### Changelog

We fixed a bug in the Chart component: Tooltip formatter callback now takes the raw data to be formatted.

## Review checklist

- [x] Description is complete
- [x] 2 reviewers (1 if trivial)
- [x] Tests coverage has improved
- [x] Code is ready for a release on NPM

### CODE REVIEW

**Comment by richterb** [comment]
I remember the use of an index was a very conscious choice for a reason, but I forgot why or if it's still a relevant reason.

The one thing I can think of would be the ability to use other data points the tooltip: for example, showing a small icon to show if the value went up or down from the previous value, or the associated label. But we could also send other parameters as well to cover all cases I guess.

**Review by richterb — APPROVED** [review]

File: packages/visualizations/src/components/Chart/Chart.svelte [hunk 1]
```diff
@@ -73,13 +82,52 @@
...
+                            if (seriesType === ChartSeriesType.Line) {
+                                return prefix + format(parsed.y) + suffix;
+                            }
+                            if (seriesType === ChartSeriesType.Radar) {
+                                return prefix + format(parsed.r);
+                            }
+                            if (seriesType === ChartSeriesType.Pie) {
+                                // For pie charts we need to get the label from the dataFrame because, unlike other
+                                // charts, the label is not the series legend, it's the category.
+                                return `${dataFrame[dataIndex].x}: ${format(parsed)}`;
```
Comment (etienneburdet) [inline packages/visualizations/src/components/Chart/Chart.svelte:126]: YES 😎

**Review by etienneburdet — APPROVED** [review]
LGTM! Hiding dataFrame[dataIndex] is really what was needed, so is the default format 👌

File: packages/visualizations/src/components/Chart/Chart.svelte [hunk 2]
```diff
@@ -4,8 +4,9 @@
     import { Chart } from 'chart.js';
     import 'chartjs-adapter-luxon';
     import type { Async } from '../../types';
-    import type { ChartOptions, ChartSeries } from './types';
     import type { DataFrame } from '../types';
+    import type { ChartOptions, ChartSeries } from './types';
+    import { ChartSeriesType } from './types';
```
[the remark this answers is not in the document]
Comment (wmai) [reply packages/visualizations/src/components/Chart/Chart.svelte:9]: I'm getting the error: `Unexpected token` by doing so 😞

**Pull request merged**